pennylane/interfaces/tensorflow.py

```diff
@@ -116,207 +165,110 @@ def _res_restructured(res, tapes):
-            If ``jacobians`` is an empty list, then ``gradient_fn`` is used to
-            compute the gradients during the backwards pass.
-        gradient_kwargs (dict): dictionary of keyword arguments to pass when
-            determining the gradients of tapes
-        gradient_fn (callable): the gradient function to use to compute quantum gradients
-        _n (int): a positive integer used to track nesting of derivatives, for example
-            if the nth-order derivative is requested.
-        max_diff (int): If ``gradient_fn`` is a gradient transform, this option specifies
-            the maximum number of derivatives to support. Increasing this value allows
-            for higher order derivatives to be extracted, at the cost of additional
-            (classical) computational overhead during the backwards pass.
+        execute_fn (Callable[[Sequence[.QuantumTape]], ResultBatch]): a function that turns a batch of circuits into results
+        jpc (JacobianProductCalculator): a class that can compute the vector Jacobian product (VJP)
+            for the input tapes.
+
+    Keyword Args:
+        device=None: not used for tensorflow
+        differentiable = False: whether or not the custom gradient vjp needs to be
+            differentiable. Note that this keyword argument is unique to tensorflow.
 
     Returns:
-        list[list[tf.Tensor]]: A nested list of tape results. Each element in
-        the returned list corresponds in order to the provided tapes.
+        TensorLike: A nested tuple of tape results. Each element in
+        the returned tuple corresponds in order to the provided tapes.
     """
-    if logger.isEnabledFor(logging.DEBUG):
+
+    if logger.isEnabledFor(logging.DEBUG):  # pragma: no cover
         logger.debug(
-            "Entry with args=(tapes=%s, jacs=%s, execute_fn=%s, gradient_fn=%s, gradient_kwargs=%s, _n=%s, max_diff=%s) called by=%s",
+            "Entry with (tapes=%s, execute_fn=%s, jpc=%s, differentiable=%s",
             tapes,
-            repr(device),
-            execute_fn
-            if not (logger.isEnabledFor(qml.logging.TRACE) and inspect.isfunction(execute_fn))
-            else "\n" + inspect.getsource(execute_fn) + "\n",
-            gradient_fn
-            if not (logger.isEnabledFor(qml.logging.TRACE) and inspect.isfunction(gradient_fn))
-            else "\n" + inspect.getsource(gradient_fn) + "\n",
-            gradient_kwargs,
-            _n,
-            max_diff,
-            "::L".join(str(i) for i in inspect.getouterframes(inspect.currentframe(), 2)[1][1:3]),
+            execute_fn,
+            jpc,
+            differentiable,
         )
 
-    # pylint: disable=unused-argument
-
     parameters = []
-    params_unwrapped = []
+    numpy_params = []
 
-    # assumes all tapes have the same shot vector
-    has_partitioned_shots = tapes[0].shots.has_partitioned_shots
-
-    for i, tape in enumerate(tapes):

```

Comment:
good thing we never used this `i`, because we re-define it _twice_ in this block 😅